src/main/java/sc/fiji/pQCT/analysis/DistributionAnalysis.java

```diff
@@ -242,42 +254,39 @@ private void calculateRadii(boolean preventPeeling) {
 					bMDJ.get(div)[et] += BMD_temp.get(ka);
 					mo++;
 				}
-				bMDJ.get(div)[et] /= mo;
+				bMDJ.get(div)[et] /= (double) mo;
 				
 			}
-			
-			//IJ.log("Sector "+et+" Endo "+bMDJ.get(0)[et]
-			//					+" Mid "+bMDJ.get(1)[et]
-			//					+" Peri "+bMDJ.get(2)[et]);
-			
 		}
 	}
 
 	// TODO Refactor into a static utility method for all classes instead of
 	// repeating code
-	private void erode(final double[] data) {
+	public static double[] erode(double[] data,int width,int height,double bgVal) {
 		// Erode algorithm
 		// Modified from the best dilate by one solution taken from
 		// http://ostermiller.org/dilate_and_erode.html
-		for (int i = 0; i < height; i++) {
-			for (int j = 0; j < width; j++) {
+		for (int i = 1; i < height-1; i++) {
+			for (int j = 1; j < width-1; j++) {
 				final int index = i * width + j;
-				if (data[index] > minimum) {
-					if (i > 0 && data[(i - 1) * width + j] == minimum || j > 0 &&
-						data[(i) * width + j - 1] == minimum || i + 1 < height && data[(i +
-							1) * width + j] == minimum || j + 1 < width && data[(i) * width +
-								j + 1] == minimum)
+				if (data[index] > bgVal) {
+					if (data[(i - 1) * width + j] == bgVal | 
+						data[(i) * width + j - 1] == bgVal |
```

Comment:
In this particular case either should result in the same result. This if should evaluate to true if any of the ... == bgVal is true. If || is faster I'd suggest that otherwise I'd leave as is. I didn't have a good reason to to change this to bitwise |, I think it just fitted on one row on my display when I took the second | off... and made it easier to inspect.

Reply:
> This if should evaluate to true if any of the ... == bgVal is true.

With the important difference that `||` will not test the follow-up statements as soon as one of them evaluates to `true`. See also [this SO answer](https://stackoverflow.com/a/1808879/1919049).

Reply:
Always use `||` for boolean logic. We actually hit a bug in Java a few years back, where some ImgLib code used `|` with booleans by accident, and it caused the Java runtime to segfault. IMHO, the java compiler should not even allow `|` with boolean operands—there is no advantage except for the non-short-circuiting @imagejan mentioned... but if you are relying on side effects in your conditional statements, the code is probably too opaque anyway.

Reply:
I was familiar with the distinction between || and | but it never occurred to me that it might not be a good idea to do a bitwise or on booleans. I just assumed false is 0x00 and true 0x01. Should not assume... On that point, I've had Java segfaults I could not figure out (occured when running with Linux but not on Windows machine) in other code I've written -> will need to check if this is the culprit. Live and learn, thank you for the comments!